### PR TITLE
Added Workload Identity setup for new and existing KFP clusters

### PR DESCRIPTION
**Pull request by ashok133**
This PR adds [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#overview) setup for new and existing KFP clusters.

### CODE REVIEW

**Comment by TheMichaelHu** [comment]
yeah, we're going to need to set up integration tests to test this consistently. I'll try to set up an example integration test in a new PR.

**Comment by ashok133** [comment]
Fixed setting up WI for existing clusters. The user flow now looks like:
1. User creates GKE cluster (using gcloud/Console/CAIP Pipelines)
1. User deploys KFP using CAIP Pipelines
1. MLPG checks if cluster has WI enabled using a custom GKE label `mlpg_wi_auth`
    1. If label is set, MLPG skips WI setup and deploys model on KFP instance. 
    1. If not set, configures WI for the cluster, migrates pods to new updated node-pool gracefully, updates the custom label and updates the KFP hostname. The user updates the hostname in config.yaml and redeploys the pipeline triggering step 3 again.

File: examples/kfp/config.yaml.example [hunk 1]
```diff
@@ -17,6 +17,8 @@
 project_id: [PROJECT ID]
 bucket_id: [BUCKET ID]
 region: "us-central1"
+cluster_name: [GKE CLUSTER NAME]
+cluster_zone: [GKE CLUSTER ZONE]
```
Comment (TheMichaelHu) [inline examples/kfp/config.yaml.example:21]: duplicate

**Review by TheMichaelHu — CHANGES_REQUESTED** [review]
Just need some integration tests, code LGTM

File: examples/kfp/config.yaml.example (hunk 1 above)
Comment (ashok133) [reply examples/kfp/config.yaml.example:21]: Those are two different params and both required to identify the cluster to apply WI for. Could you clarify further?

Comment (TheMichaelHu) [reply examples/kfp/config.yaml.example:21]: oh oops, misread sorry

**Comment by TheMichaelHu** [comment]
Alternatively, lmk if you want me to write the tests and we merge this in first.

**Review by TheMichaelHu — APPROVED** [review]
LGTM, we can add tests either in this PR or a future one.

**Comment by ashok133** [comment]
Better to have tests in a separate PR.

**Pull request merged**